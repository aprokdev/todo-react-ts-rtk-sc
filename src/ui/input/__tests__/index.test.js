import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act, create } from 'react-test-renderer';
import Input from '../index';

const props = {
    checked: false,
    className: 'test',
    id: 'test',
    onBlur: () => null,
    onChange: () => null,
    placeholder: 'Type...',
    type: 'text',
    value: '',
    testId: 'input',
};

describe('Input', () => {
    test('matches snapshot', () => {
        let tree;
        act(() => {
            tree = create(<Input {...props} />);
        });
        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('add passed className prop to the end of classlist', () => {
        const className = 'test';
        render(<Input className={className} testId="input" />);
        const component = screen.getByTestId('input');
        const classListLength = component.classList.length;
        const lastClassName = component.classList[classListLength - 1];
        expect(lastClassName).toBe(className);
    });

    test('set default "text" type attribute if none was passed to the component', () => {
        render(<Input testId="input" />);
        const component = screen.getByTestId('input');
        const typeAttr = component.getAttribute('type');
        expect(typeAttr).toBe('text');
    });

    test('onChange event changes value', async () => {
        const handleChange = jest.fn();
        const user = userEvent.setup();
        render(<Input onChange={handleChange} testId="input" />);
        const input = screen.getByTestId('input');
        await user.type(input, 't');
        expect(handleChange).toHaveBeenCalledTimes(1);
        expect(input.value).toBe('t');
        await user.type(input, 't');
        expect(handleChange).toHaveBeenCalledTimes(2);
        expect(input.value).toBe('tt');
        await user.clear(input);
        expect(handleChange).toHaveBeenCalledTimes(3);
        expect(input.value).toBe('');
    });
});
