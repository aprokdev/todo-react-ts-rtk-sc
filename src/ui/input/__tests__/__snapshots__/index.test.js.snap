// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<input
  className="input test"
  data-testid="input"
  id="test"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Type..."
  type="text"
  value=""
/>
`;
